txpool/txpool.go

```diff
@@ -64,126 +76,122 @@ type signer interface {
+	// map of all accounts registered by the pool
+	accounts accountsMap
 
-	// Max price heap for all transactions that are valid
-	pendingQueue *txPriceHeap
+	// all the primaries sorted by max gas price
+	executables *pricedQueue
 
-	// Min price heap for all remote transactions
-	remoteTxns *txPriceHeap
+	// lookup map keeping track of all
+	// transactions present in the pool
+	index lookupMap
 
-	// Lookup map that keeps track of txns present in the pool
-	txnLookupMap map[types.Hash]*types.Transaction
-
-	// Lock for the txn lookup map
-	txnLookupMapLock sync.RWMutex
+	// networking stack
+	topic *network.Topic
 
-	// Gauge for measuring pool capacity
+	// gauge for measuring pool capacity
 	gauge slotGauge
 
-	// Networking stack
-	topic *network.Topic
+	// channels on which the pool's event loop
+	// does dispatching/handling requests.
+	enqueueReqCh chan enqueueRequest
+	promoteReqCh chan promoteRequest
 
-	// Flag indicating if the current node is a sealer,
+	// shutdown channel
+	shutdownCh chan struct{}
+
+	// flag indicating if the current node is a sealer,
 	// and should therefore gossip transactions
 	sealing bool
 
-	// Flag indicating if the current node is running in dev mode
+	// flag indicating if the current node is running in dev mode (used for testing)
 	dev bool
 
-	// Whether local transaction handling should be disabled
-	noLocals bool
-
-	// Addresses that should be treated as local
-	locals *localAccounts
-
-	// priceLimit is a lower threshold for gas price
-	priceLimit uint64
-
-	// Notification channel used so signal added transactions to the pool
-	NotifyCh chan struct{}
+	// prometheus API
+	metrics *Metrics
 
-	// Indicates which txpool operator commands should be implemented
+	// indicates which txpool operator commands should be implemented
 	proto.UnimplementedTxnPoolOperatorServer
-
-	metrics *Metrics
 }
 
-// NewTxPool creates a new pool for transactions
+// Returns a new pool for processing incoming transactions.
 func NewTxPool(
 	logger hclog.Logger,
-	sealing bool,
-	locals []types.Address,
-	noLocals bool,
-	priceLimit uint64,
-	maxSlots uint64,
 	forks chain.ForksInTime,
 	store store,
 	grpcServer *grpc.Server,
 	network *network.Server,
 	metrics *Metrics,
+	config *Config,
 ) (*TxPool, error) {
-	txPool := &TxPool{
-		logger:        logger.Named("txpool"),
-		store:         store,
-		idlePeriod:    defaultIdlePeriod,
-		accountQueues: make(map[types.Address]*accountQueueWrapper),
-		pendingQueue:  newMaxTxPriceHeap(),
-		remoteTxns:    newMinTxPriceHeap(),
-		txnLookupMap:  make(map[types.Hash]*types.Transaction),
-		gauge:         slotGauge{height: 0, limit: maxSlots},
-		sealing:       sealing,
-		locals:        newLocalAccounts(locals),
-		noLocals:      noLocals,
-		priceLimit:    priceLimit,
-		forks:         forks,
-		metrics:       metrics,
+	pool := &TxPool{
+		logger:      logger.Named("txpool"),
+		forks:       forks,
+		store:       store,
+		idlePeriod:  defaultIdlePeriod,
```

Comment:
We can remove idle period from the txpool, as it's not used anywhere

Reply:
[Removed](https://github.com/0xPolygon/polygon-sdk/pull/287/commits/355a0b10c85512a4f95eed8bd4a0910f7c377dfa)